packages/next/src/utilities/getPayloadHMR.ts

```diff
@@ -50,6 +53,17 @@ export const getPayloadHMR = async (options: InitOptions): Promise<Payload> => {
   if (cached.payload) {
     const config = await options.config // TODO: check if we can move this inside the cached.reload === true condition
 
+    // Generate types on startup
+    if (config.typescript.autoGenerate !== false) {
```

Comment:
This needs to be moved into the `reload` function. It should only happen if Payload is reloaded

Reply:
This would prevent generating the types on dev startup, though. Wouldn't we want this to get generated initially as well?

Reply:
the same thing should also be called in `payload.init`